base_local_planner/src/obstacle_cost_function.cpp

```diff
@@ -110,12 +112,11 @@ double ObstacleCostFunction::scoreTrajectory(Trajectory &traj) {
     return -9;
   }
 
-  std::vector<geometry_msgs::Point> scaled_footprint = getScaledFootprint(traj);
-
    const unsigned int point_size = traj.getPointsSize();
   // ignore first trajectory point since it is the same for all trajectories,
   // unless trajectory only contains one point
   for (unsigned int i = point_size > 1 ? 1 : 0; i < point_size; ++i) {
+    const auto scaled_footprint = getScaledFootprint(traj, i);
```

Comment:
Q: Somewhat related to the other thread: Do we have / use some tooling to assess effects on performance? E.g. do we know how `getScaledFootprint` compares to `footprintCost`, and how `ObstacleCostFunction::scoreTrajectory` compares to other cost functions?
I.e. if this scoring function and within it `getScaledFootprint` are what takes most time, `getScaledFootprint` should maybe pre-compute the footprint for e.g. 100 scale values between 0 and 1?

Reply:
Or actually, could we just pre-compute the footprint for the scale values 0 and 1, and then just do `scaled_footprint[i] = scale *  scaled_footprint_scale_1[i] + (1-scale) *  scaled_footprint_scale_0[i]`? That would be simple and very fast!

Reply:
> Do we have / use some tooling to assess effects on performance?

We have a task this sprint to do exactly that! For some reason DWA is ~10 times slower than TEB so we want to figure out why and fix if possible

Maybe let's leave these suggestions for optimization on hold until we can properly profile dwa? 
I think there's no point in making the code uglier if it wouldn't make a difference on performance in the end